benchmarks/flatten.js

```diff
@@ -0,0 +1,38 @@
+const faker = require('faker');
+const pull = require('../');
+
+const bench = require('fastbench')
+
+faker.seed(24849320);
+function getRandomValues(nestLevel) {
+  return Array(faker.datatype.number(1000) + 1)
+  .fill(1)
+  .map(e => faker.datatype.number());
+}
+
+const N=100;
+const randomValues = Array(N)
+  .fill(1)
+  .map(e => pull.values(
+    new Array(500)
+    .fill(1)
+    .map(e => pull.values(getRandomValues()))
+  ));
```

Comment:
This seems like a pretty heavy data stream structure. Is this common in eg jitdb?

Reply:
If I'm [reading this code](https://github.com/ssb-ngi-pointer/jitdb/blob/master/operators.js#L454) correctly, JITDB uses `flatten` to convert pages of results into individual results in `toPullStream`.

I think it also uses `flatten` to [combine live streams with log streams](https://github.com/ssb-ngi-pointer/jitdb/blob/master/index.js#L1244).

Correct me if I'm wrong @staltz @arj03, you know way more than I do about JITDB.

If I'm guessing right, then the most realistic usage would be flattening long streams of small object arrays and combining a couple very long streams of objects. @mixmix do you have a particular preference for the kind of data structures used in this benchmark?

Reply:
Ah, I see now how you are reading this code. `flatten` actually does a time-abstraction flattening, not a data structure flattening. So there isn't a stream of pages converted to a stream of items, there's a stream of stream of items converted to a stream of items.

Reply:
Maybe I shouldn't have said data structure. I think I understood how it was working, it was more understanding of this magnitude of streams was really going to be likely in the wild.